### ISSUE TITLE
Feature request: do not show "conversations" with unaccepted buddies (spam)

### ISSUE THREAD

**Issue by crosser** [iNPUTmice/Conversations#2608]
* **Version:** 1.20.1

XMPP spammers send messages and subscription requests. Messages from "non-buddies" can be blocked on the server using a privacy rule to only accept messages from the users in the roster. You still see subscription requests, that you have to deny by hand (or ignore) but this is a smaller problem.

However conversations create empty "conversations" with these spam users, that pollute the main screen that shows the list of active conversations.

It would be nice if such empty conversations with unaccepted users did not show up in the list.

**Comment by iNPUTmice** [comment]
I'm aware of the current spam wave using subscriptions but I don’t think just hiding subscription requests is a good solution to that. How would you see legitimate requests?

**Comment by crosser** [comment]
Well, ideally, I'd want to see an alarm notification (like a grey bell that turns red) in the corner meaning "pending subscription requests" that, when tapped, would open the list of pending requests. And a new "conversation" only when there is a message (other than the sub request) present. Something like that?

**Comment by iNPUTmice** [comment]
> And a new "conversation" only when there is a message (other than the sub request) is present. Something like that?

I've seen spam that sends both a message and a subscription request.


I'll leave this open for the separate 'Pending Subscription Requests and Invite' screen idea but it won't have a priority as this will do very badly to fight spam.

Client side UX changes are not the right approach on fighting spam. Spammers shouldn't dictate on how we want to design our UI.

**Comment by crosser** [comment]
I agree, but I think that a separate screen for invitations is a good idea regardless of spam.

message + invitation spam is exactly the case that I complain about. I can filter messages on the server side, but I do not want to filter invitations. So, it is the UI's job to make invitations unobtrusive. Pidgin's UI does that pretty well.

I understand that it's low priority, just wanted to throw in this idea.

**Comment by borsti67** [comment]
I'd like to add a remark regarding the "Privacy Please"-plugin for Pidgin. It offers a simple challenge-response mechanism: you define a question/answer to automatically send to unknown senders. Only after they passed this test I will see the following messages, otherwise they will see the question again.
For my home PC this works excellent (at the moment I have very few invitation spam and Pidgin doesn't show a popup) but on the smartphone it is extremely annoying. Also all of these spam conversations mess up the screen, one needs to "end conversation" for all of them even if not reading it...
Maybe something like that could be implemented?

**Comment by crosser** [comment]
> Also all of these spam conversations mess up the screen, one needs to "end conversation" for all of them even if not reading it...

You can slide the to the side without reading. But yeah..

**Comment by splurched** [comment]
>  using a privacy rule to only accept messages from the users in the roster

XEP-0016?  🏆

**Comment by borsti67** [comment]
@splurched that would mean you cannot accept/invite new contacts.
